src/plugins/announce.lua

```diff
@@ -0,0 +1,53 @@
+-- URL announcements - follows http->https redirects but not www
+
+local https = require 'ssl.https'
+local ltn12 = require 'ltn12'
+
+local plugin = {}
+
+plugin.help = 'Usage: announce <url>'
+
+plugin.main = function(args)
+    local _, _, url = args.message:find('(.+)')
+
+    if args.modules.config.debug then
+        print(url)
+    end
+
+    if not url then
+        args.modules.irc.privmsg(args.target, ('%s: Please give me a url.'):format(args.sender))
+        return
+    end
+
+    if url:find('[ <]?https?://[^> ]+.*') then
+        url = url:match('[ <]?https?://[^> ]+.*')
+    elseif url:find(' www%..*') then
```

Comment:
why the space before `www`?

Reply:
in case someone does something like `announce It's my homepage www.myhome.com`

Maybe its not likely, I have no idea XD edge cases heh